ui/components/multichain/pages/gator-permissions/components/review-gator-permission-item.tsx

```diff
@@ -42,17 +42,72 @@ import {
 } from '../../../../../../shared/lib/gator-permissions';
 import { PreferredAvatar } from '../../../../app/preferred-avatar';
 import { BackgroundColor } from '../../../../../helpers/constants/design-system';
-import {
-  getNativeTokenInfo,
-  selectERC20TokensByChain,
-} from '../../../../../selectors/selectors';
-import { getTokenMetadata } from '../../../../../helpers/utils/token-util';
 import { getPendingRevocations } from '../../../../../selectors/gator-permissions/gator-permissions';
+import { useGatorPermissionTokenInfo } from '../../../../../hooks/gator-permissions/useGatorPermissionTokenInfo';
+import { CopyIcon } from '../../../../app/confirm/info/row/copy-icon';
+import { Skeleton } from '../../../../component-library/skeleton';
+
+// Shared row style for permission details
+const rowStyle = { flex: '1', alignSelf: 'center' } as const;
 
-type TokenMetadata = {
-  symbol: string;
-  decimals: number | null;
-  name: string;
+type PermissionDetailRowProps = {
```

Comment:
nit: should we export this - given it's the params type for the exported component.